api/applications.go

```diff
@@ -37,15 +39,103 @@ type ApplicationListResponse struct {
 	Icon        string `json:"icon"`
 }
 
+// DeleteApplications delete application
+func DeleteApplications(c *gin.Context) {
+	idParam := c.Param("id")
+	id, err := strconv.ParseUint(idParam, 10, 32)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, components.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: fmt.Sprintf("Invalid id=%q", idParam),
+			Error:   err.Error(),
+		})
+		return
+	}
+	// Build application model TODO refactor
+	db := model.GetDB()
+	app := &model.Application{
+		ID: uint(id),
+	}
+	var deployments []*model.Deployment
+	db.Model(app).Related(&deployments, "Deployments")
+	log.Debugf("Associated deployments: %#v", deployments)
+	organization := auth.GetCurrentOrganization(c.Request)
+	err = db.Model(organization).Related(app).Error
+	if err != nil {
+		log.Errorf("Error getting application: %s", err.Error())
+		c.JSON(http.StatusBadRequest, components.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: "Error getting application",
+			Error:   err.Error(),
+		})
+		return
+	}
+	app.Deployments = deployments
+
+	// Get cluster for application
+	filter := make(map[string]interface{})
+	filter["organization_id"] = organization
+	filter["id"] = app.ClusterID
+	commonCluster, ok := GetCommonClusterFromFilter(c, filter)
```

Comment:
Not clear why we need to use `GetCommonClusterFromFilter` here.
There is already a FK constraint on `Application.ClusterId` field, so we should be able to retrieve the Cluster that the app belongs to based on `app.ClusterID` only. Once we have the cluster db record we can create the common cluster from it.



I'd add a foreign key constraint to `Application.ClusterID` and retrieve the related cluster based on the FK form db.  The FK would also